### PR TITLE
Fix the simulation API not showing detail error info

### PR DESCRIPTION
**Pull request by areshand**
## Description

we try to get the detail error from VM status directly, the detail error in txn output may be lost after feature turned on

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

Added new test cover simulation API detail error info

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: api/src/tests/transactions_test.rs [hunk 1]
```diff
@@ -1504,6 +1504,44 @@ async fn test_simulation_failure_error_message() {
...
+    let txn = raw_txn
+        .sign(invalid_key.private_key(), account.public_key().clone())
+        .unwrap()
+        .into_inner();
+    let body = bcs::to_bytes(&txn).unwrap();
+    let resp = context
+        .expect_status_code(200)
+        .post_bcs_txn("/transactions/simulate", body)
+        .await;
+    assert!(resp.as_array().unwrap()[0]["vm_status"]
```
Comment (runtian-zhou) [inline api/src/tests/transactions_test.rs:1539]: Will this test fail on main without your patch?

**Review by runtian-zhou — COMMENTED** [review]
Great to have a persistent test for this. It would be great if such failure could be captured by CI in the future.

File: api/src/tests/transactions_test.rs (hunk 1 above)
Comment (areshand) [reply api/src/tests/transactions_test.rs:1539]: it fails. This the resp without the change     "vm_status": "Execution failed with miscellaneous error and no status code\nExecution failed with message: Linker Error: Module 0000000000000000000000000000000000000000000000000000000000000001::MemeCoin doesn't exist\nBacktrace: . And the resp with the change `vm_status":"LINKER_ERROR\nExecution failed with message: Linker Error: Module 0000000000000000000000000000000000000000000000000000000000000001::MemeCoin doesn't exist\nBacktrace: Backtrace `

Comment (runtian-zhou) [reply api/src/tests/transactions_test.rs:1539]: Beautiful!

**Review by runtian-zhou — APPROVED** [review]

**Review by msmouse — APPROVED** [review]

**Pull request merged**